### ISSUE TITLE
Symbol is incompatible with index signature

### ISSUE THREAD

**Issue by DeTeam** [Microsoft/TypeScript#21962]
I'm trying to use both the index signature in my interface and some symbol  fields. They don't seem to work very well together.

**Code**

```ts
export const KEY = Symbol('KEY')

export interface Tree {
  [KEY]?: string
  [name: string]: Tree
}

// Here I get a type problem
export const tree: Tree = {
  'a': {
    [KEY]: 'value1',

    'x': {
      [KEY]: 'value2'
    }
  },
  'b': {
    [KEY]: 'value3'
  },
  'c': {
    [KEY]: 'value4'
  }
}
```

**Expected behavior:**

I think there're two options here:

- Either complain about things on the type level (like if one'd use `key` (just a string field name) instead of a symbol `[KEY]`)
- Or allow such usage together with access later `const s: Tree = tree['c']; const s: string = tree['c'][KEY]`

**Actual behavior:**

> Property '[KEY]' is incompatible with index signature.

**Playground Link:**  

**Related Issues:**

I think that maybe the following ones are related in a way:

- https://github.com/Microsoft/TypeScript/issues/21912
- https://github.com/Microsoft/TypeScript/issues/1863
- https://github.com/Microsoft/TypeScript/issues/20146

**Comment by yortus** [comment]
I think #21217 was meant to fix this, but this shows there is still a problem. Simpler repro:

```ts
export const SYM = Symbol('a unique symbol');

export interface I {
  [SYM]: 'sym';
  [x: string]: 'str';
}

let a: I = {[SYM]: 'sym'}; // ERROR: 'sym' is not assignable to 'str'
let b: I = {[SYM]: 'str'}; // ERROR: 'str' is not assignable to 'sym'
```

**Comment by mhegazy** [comment]
@weswigham i would have expected this to be fixed by #22339, but does not seem to be the case..
